### PR TITLE
Phase One

### PR DESCRIPTION
**Pull request by PTNielsen**
Completed work from the Phase One Trello Card up to the point of generating unique id's for each correctable field and saving questions in our database, per our prior conversations.

The original application has been set up to run at `/phase_two` and an additional endpoint, `/phase_one`, has been created to allow a user to complete blank math quizzes.  The root of the application has been set to redirect to `/phase_two`.  This was mainly accomplished by updating `config/routes.rb` to utilize an optional route scope so we can re-use the existing controllers without having to duplicate with namespaces.  We determine which view files to render by referencing `params[:phase]` from each request.

The `Response` model has been updated to save which phase the response was recorded under.  We will need to run a migration for this to take affect and the test database must be updated as well.

### CODE REVIEW

File: app/assets/javascripts/save_functions.js [hunk 1]
```diff
@@ -19,8 +19,11 @@ window.send_correction = function(
     current_value,
     correct_answer,
     quota,
-    round_number
+    round_number,
+    phase
 ) {
+    phase_int = phase === 'phase_one' ? 1 : 2
```
Comment (rposborne) [inline app/assets/javascripts/save_functions.js]: Is there anyway we could make this less "magic numbery"?  Is phase_int, the backend phase ID?

File: app/views/phase_two/options/2/1_1.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<h2>Math Quiz  1: Contains X Errors</h2>
+<h2>Math Quiz  1: Contains X Errors 1_1</h2>
```
Comment (rposborne) [inline app/views/phase_two/options/2/1_1.html.erb]: Did we intend to change this?

**Review by rposborne — CHANGES_REQUESTED** [review]
@PTNielsen I have done a first pass of "code" review, a few minor tweaks but nothing major.  I think we need to get on the phone with @econglet in regards to the concerns around programmatic ids and tracking who answered when a specific version of a question.

For reference
We can very easily generate ids, but it becomes an issue when these questions/essays change over time.  It would be easy to move the math questions into a controlled sense, where they each have formal id's and you edit them one by one through the interface (vs code)  but this breaks down when we start thinking about essays.   We could also SHA the question or essay to at least have a fingerprint but this will not be meaningful for qualifying if the change would change the scientific results. i.e., does changing a comma or removing an extra space change the data for the report.

File: app/views/phase_two/options/2/1_1.html.erb (hunk 2 above)
Comment (PTNielsen) [reply app/views/phase_two/options/2/1_1.html.erb]: Had that in there to ensure I was getting the correct view files to render.  Oversight on my part in not removing it - good catch!

File: Gemfile [hunk 3]
```diff
@@ -20,7 +20,7 @@ gem 'airbrake'
 gem 'bootstrap-sass', '~> 3.3.3'
 gem 'sass-rails', '~> 5.0'
 # Use Uglifier as compressor for JavaScript assets
-gem 'uglifier', '>= 1.3.0'
```
[the remark this answers is not in the document]
Comment (PTNielsen) [reply Gemfile]: I believe we discussed this a while back when I was initially getting my dev environment set up to begin work on this though I can't recall why exactly the change was made.  I'll revert.

File: app/assets/javascripts/experiment_selection.js [hunk 4]
```diff
@@ -1,5 +1,8 @@
 //this is :(
 jQuery(function() {
+    // window.phase is set in phd.js but the below line is just in case
+    // experiment_selection.js is ever loaded without phd.js for whatever reason
+    window.phase = window.phase || window.location.href.match(/phase_[a-z]+/) ? window.location.href.match(/phase_[a-z]+/)[0] : '';
```
[the remark this answers is not in the document]
Comment (PTNielsen) [reply app/assets/javascripts/experiment_selection.js]: Done.

File: app/assets/javascripts/phd.js [hunk 5]
```diff
@@ -1,14 +1,15 @@
 var TIME_LIMIT_IN_MINUTES = 7;
 var INTERVAL_IN_SECONDS_OF_HOW_OFTEN_TO_SHOW_OTHER_STUDENT_ACTIONS = 60; // one minute
 var TIMEOUT_FOR_OTHER_PARTICIPANTS = 15 * 1000;
-var NUMBER_OF_ROUNDS = 4;
+window.phase = window.location.href.match(/phase_[a-z]+/) ? window.location.href.match(/phase_[a-z]+/)[0] : '';
```
[the remark this answers is not in the document]
Comment (PTNielsen) [reply app/assets/javascripts/phd.js]: Also done

**Review by rposborne — APPROVED** [review]

**Pull request merged**